obj_factory.h

```diff
@@ -102,6 +102,42 @@ class ObjFactory {
         return vec[idx.val].get();
     }
 
+    bool exist(const std::string& uri) {
+        if (map.find(uri) == map.end()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    bool erase(const Idx<ObjType>& idx) {
+        if (idx.val >= vec.size()) {
+            return false;
+        }
+        // erase map member
+        const auto* elem_ptr = vec[idx.val].get();
+        if (map.erase(elem_ptr->uri) == 0) {
+            return false;
+        }
+        // erase vec member
+        vec.erase(vec.begin() + idx.val);
+        return true;
+    }
+
+    bool erase(const std::string& uri) {
+        if (exist(uri) == false) {
+            return false;
+        }
+        const auto* elem_ptr = map.at(uri);
+        // erase map member
+        if (map.erase(uri) == 0) {
+            return false;
+        }
+        // erase vec member
+        vec.erase(vec.begin() + elem_ptr->idx);
```

Comment:
I guess the re-Indexation also applies here ;)

Reply:
yes